test-d/union-to-tuple.ts

```diff
@@ -0,0 +1,6 @@
+import {expectAssignable} from 'tsd';
+import {UnionToTuple} from '..';
+
+declare type Union = 'A' | 'B';
+declare const tuple: UnionToTuple<Union>;
+expectAssignable<readonly ['A', 'B']>(tuple);
```

Comment:
I think we need to extend these tests quite a bit. This only tests a "sunny day scenario", the simplest of cases, we should have some tests that tries to break it.

I eg. tried this:

```typescript
type AdvancedUnion = 'create' | number;

type AdvancedTuple = UnionToTuple<AdvancedUnion>;

const foo: AdvancedTuple = ['create', 123];
```

And that fails (see [playground](https://www.typescriptlang.org/play?#code/PQKhCgGEHsDsDcCmAnALgAgIboK6wJZzqoCeADosdFrOvrKigM6IDGqhtpFuT9A5ugDaAE3xNUyfACMcHJOlZwxHOJgA2xcoiYBdABQALVKjJMAXMGAB3WwDpuO1lLKp1mWPzvRk-YCOhWJmBDDxFpaGgAa2BkRHVETBYAWlhoRmDHJmd8V2SAJmSADjtjAFt1AGIxCSlZeURkpVgVTg1krIBKO3BwAElYJjJ8OJF0aRJhVENxdABlVExWKPQAeSRkADN1aGsaJmsUA2NTCysJJajoDe3duyUy4ExgAFYABgBmAHYX-KKANmA+QAjF8QS8Pt1egABRAAD0wZTICXAAAN0eB8EifBgAN4AVQIcAAKtABoxkCx2JwAL7oTbIaBldAAckcyU2OlQLIA3L1HOhCZx0ABedC46aIfSdczoeDQfAiOkAH3F-DimFQ+kwvlltQEMrlCqV6FVuKcmAosoi0ASHhpfPAAvJzDYqloYqFJLJDFd1LgAB4vbAAHx8qwikPiyXS2XyxU89DqxJanX8PWSA1x42Ji1W8aRO2wB1ojHgACC6DKPkoluR+FYmGkCXQ8MRyMo1hmrEMimgOHUYzKmCilHwqCY6GsmEm9FQ1BI-eQ6HUMmQOsmShElE2tp21iYPXAsIRSJR6NRmOxaHFwdJLspbtp9MZzLZ2g5XN5vVY7iYk5gMphxaVZYEoXFwAASAeYCRAscUoMgzBgVlaVRSjPBt02ehEBEAAaRDpBQ9A0MjXAWkQbCwPwqCaXAOjwF-JIAKZWDiWsagIOg1iwngrikPyVC03LDMpE8Tp0PIrCcJoyDIOkQSSLTAAhUSDUkzDKJkgjILohjmgkcjhTFIQwL2QDYNAqVOjw9AzPQCywnY6BpV0OxhzIfR6AuWBWEoMjvMWXzEHuHiWiYTo+QFYNRS0ChoE2Iy4HDYAyNxZDYyNBNxmBTL4xNM1MHybVfBE9B9XE7NsoUkr+FU8rM0qrKlX5bR0AfKl3Viu8fQpTrOCDIlQxStKMsNfLEyIvKcywYrhLUpqJvGObfHqir+HG406IvcAQGAcB4TIHE4soHqOqfQNgyjMV9Cgqx0FReFGHC8iojSaxYFROhJw0acSEnfgFU8KhxkoSVFCSWsWm+3hcJB5oNgwSU7uAB7gy+udqGwURxEzep8AUZpWjUdQUZO44TDMSwbHsLIclcdxPG8Xx-ECYJQhaG0YjiO0UjSDJgDplxUAKYpSlQCpqlxuo5AJxoifHNp1A6bQIp6SCYqexAXrwN7dlgRD7uJQxKDwYUBVmHAWDGJJiBN9A4HUSY0xwMptaRrH6Twf1aD4YK7dNobDdRhKvd8rqXbdhhJ1mDw6F9R8ffVyCAH4SJqPG5Fw4NZWDCSyPy4P0GN2YONgNl0FCMgKFgZPZTAjYi4GTll3B1Fzp9r6BT9vzi+0OZ6YwOIyDiFho4D9Ajr4d0NCLyP3cnUOzbgRfEs2b33R+n7aDnP0utD8Hl9rqCJK1l79H0N3fFwjv3VlegW-ahP+rgfOo3yzpELT2-OEQ+vEA2I6UAEAYAIBQBgOOh1VysHHElLgbV5xYGIDgDsPRjaUHiIgKOGAfDbmXAfe2qAUEtjCA7RKh8hpTnwOoTQ0gwb2yYIiEK6BlJsEwFbSgBDNS2UXDgcqhh+yDjsukWGdtZj4HIfbI+MMHhkCzmMQKiQxih2AjgDQTt0DbiYqMWydDGwcInsSfug8+xImoSgTR0AdDCIwPwNR65fT0M1BPXBFjQ5MBwD2OBrYEjYJjlwUIGBGwsB6AMIYIw4YTCmDMGO-4cBjgCZQIxFAB7C3QMPaAFNTjU34OOQwOBpChUeGUBsjImAJVQMAZJiBUm5EqeIDxOhgDAg+PkAAnEUKE+IUjBMQLKAAmv2OyiA4aIICHKDQipNTCnqOgXhljy4YFCAoSBsBXa2Uds7Wg0jHB2HmFiahOoQbgyEKiAAonCSQSxUBfTbGeBJE8NQiDdlkqmVghZ1OSMiGcdgABWh4fB+BTosa+qART-EqFuRAwBID9PxNYOE-AABy-BDD8GJPgcsZzywAC9yy+FWHigAYtYAAUv0gAEqgAAGjiuY1gzkiCiAARWUuS8skB+BnOgOWeJAA1AAWuWAA6tYSAABZZl-TlI4HLAAaQACIKusAAcVYOKoo4q6pEoACzlmpZ4ZS6hyykv+NkM5cx+AUr6AAJXwP8UgLwBX4hVQEfAaLsXKWpQKoovryxIpeCIHVRR-UKl5fAfEmp8gfDlf8d1kq2nlmJGct4ykcV8o+IDAACiQIoKqBW-IFasAAMnVTYxb8B6sQL835eqzlnL6JAZS1hywiBxS8HV4ryzlmLXKolzLmUAGplJwjOYgbtFL+nAhRXK8sHE+hypIG8dUyAdUKuUiQM5+IyD8FYKwHd4rMDUv4OWTAcJWHIEgDi-Aw62nJAFQOEg-AFX9iJRSm1TB8TAhxUUKIcrxX9LOcCG1oQACOJ6KVRFQHqyDwqOXQexeWLN0g+U2v4MW4twqUXKTIG0-gKr1DKSYEUHAzLMD-BIOWfE+IcD9Kzcy6wIhkVkH-V8WAhgvhlHFeK4tOrOXlkHTal4cHlKgfxOWfgZAFWID5XCHAhg4SrGBCbN4-zqUqsMEUblqxlKrDaZAVYcIPjKWZVm6A+Qs3KXLAq5YmAbX-D6L8qIcxyz-Es6B4k-AmDirTVmvgrARDIDOZAL4qxSXCupXMZlSLrD9PgESs5J6+XEjmG00lxbqWrAVRQbLcrlICqJfwHVKqYDCswEUPlHK+ikpIFEeV+BmXCsXfkfpzLVjlkIP0-AUROVvFWGc8VCqSCGBtXMEga6SBwjmPkHVcrDDdvFb8kg9r+BzBDXO35hgXhZbePgZSb7yxRFJdAM5qwohMEgMCMosAojitJaBoo-B8TMsVUikwcqs3qFgAyWLZAPikpeNSzYykKXUtA9AOVZBNj9Lq6sPoOqPBMGJAiRACrrPVlG5AJt-TyzSDhIYSA-xiQKrhP8ZAsXrU2upS8aQA9fnlh1dIYtpKOIKrKMqhVURNjSHLGl8VyA+Xir5SIcVwJiRIv+G00tURPPWH4FEHskbDCgb6OWGy4w5Dxzts43pE9UQ9g8PwRAWbmBwAVZqTAX117h2FAEax-MJmrhEJqehlBUSjhIF9MgOomEUiwEPPAHA3a2WHFEAQdAcGwA0Ywzk8c+Dbj7ikweezunh+wEQjsNAxionxeuEglR6C-hwNuJgVufCO6mV1XJSBJxW3SYHrElAY+IFIPScv0xnFh+hqHa3Sdjx3I7KWS80VKFihZOWFkppWTKW-AZJGxC+mCiGqSYkC-BqcGusIcfLJbIshn7oPkO0TztnPBiI2MSTpVhnKYncr4kGrmkOuZAkwO8YHmTqU2fsvBOjauWH2TABg5gmREAK1H8NxYoWRnAUxEBJ9VQWQnlYDWQUFndGBECWRtwEhUCp8WRMB+Bkx+AXdvwBRywaFdhcI-9N5YoLki9twAwKDOAADYAgC3ZQCn8SBd9cD8DCCwxGIV4IF-9c5l9oBV8Ow6DSDDgRB6CV5N8hBICNRUDd8EDd9kDCDd8MCW8YCD9wBrkGC7BC91Bi8dB9A4hQN4kJByCfYJJ7o+VJkUDEAh9dp9pDpjoR9OAV818rpRRwB0AfCl83Deo94BpvDfCQj7oFVpYZBZZCZlBFYSZiApBlgegQjQjUY5gRkJ5ykcBkBe5dxW57ZUQzpn4LpPpL9ciqwaxNEW9MBqFDxgjki-CiAz44JXp3oDZ6ifC059AM4ZZGARBBDOA35mo6jkj-5G4QiowmjJwL4r4jc+j45H4f5X5JIP5hiUji4HEmBciygpjt9sCZ8JJMZhAhAhBdBbIAAicsM4049AM45SK43RTXEvSQTxIhUeCeHUPPJgVY3wtOJ5TZYQOwQEnqEQhIAMaggw2g4MWyRY0MEMaEoon2XQb4nw2UE4oBVGEuScAUYcTcYAl8Jke-NccA1-OZIZD-XgAQH-HgVteADwPyEQGKMfKAwgqfNZMoOhZAR0YgkQWk4KEQEE-yBo2Adw0Qmkuk7OIaHg3gwYDAXcaAWUMUvkgU2KWQ5khQ9AYEaNLQoAA)) as for some reason the calculated tuple becomes `[number, 'create']` rather than the expected `['create', number]`.

I similarly tested `UnionToTuple<'create' | boolean>`, which results in `[false, true, 'create']`, so also it returns things in the wrong order, but it also extends `boolean` into `false` and `true` respectively.

If `UnionToTuple` only supports `string` unions, then it should be changed to `UnionToTuple<Union extends String>` + a test should be added with `expectError()` that ensures that it eg. `UnionToTuple<'create' | boolean>` then fails.

Alternatively, if `UnionToTuple` is supposed to support more than just strings, then tests should be added to ensure that + the implementation fixed so that it works as expected.

I would also like a test for:

```typescript
type Foo = 'abc' | 'def';
type Bar = Foo | 'xyz';
type GiveMeThatTuple = UnionToTuple<Bar>;
```

Which would highlight if the expected outcome is `['abc' | 'def', 'xyz']` or `['abc', 'def', 'xyz']`. It may be obvious, but yet not obvious.